tests/APIv2/UsersTest.php

```diff
@@ -129,6 +129,37 @@ public function testDeletePhoto() {
         $this->assertStringEndsWith('/applications/dashboard/design/images/defaulticon.png', $user['photoUrl']);
     }
 
+    /**
+     * Test confirm email is successful
+     */
+    public function testConfirmEmailSucceed() {
+        $emailKey = ['emailKey' => 'test123'];
+        $user = $this->testPost();
+
+        /** @var \UserModel $userModel */
+        $userModel = self::container()->get('UserModel');
+        $userModel->saveAttribute($user['userID'], 'EmailKey', 'test123');
+
+        $response = $this->api()->post("{$this->baseUrl}/{$user['userID']}/confirmEmail", $emailKey);
+        $this->assertEquals($response->getBody(), ['emailConfirmed' => true]);
+    }
+
+    /**
+     * Test confirm email fails
+     *
+     * @expectedException \Exception
```

Comment:
This should be more specific. How would you know if this exception was thrown because of the bad email confirmation key or if it was thrown because of a random error in the dispatcher? Right now, all this test confirms is that something went wrong. You could use the `@expectedExceptionMessage` tag for greater specificity.

Reply:
resolved